dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -132,14 +138,17 @@ protected virtual void Dispose(bool disposing)
     // HTTP user agent sent to remote endpoints
     private const string HTTPUseragent = "Microsoft Semantic Kernel";
 
-    private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody)
+    private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody, CancellationToken cancellationToken = default)
     {
         string responseJson;
 
         try
         {
             using HttpContent content = new StringContent(requestBody, Encoding.UTF8, MediaTypeNames.Application.Json);
-            HttpResponseMessage response = await this.HTTPClient.PostAsync(url, content);
+
+            HttpResponseMessage response =
+                await this._retryPolicy.ExecuteWithRetryAsync(async () => await this.HTTPClient.PostAsync(url, content, cancellationToken), this.Log,
```

Comment:
nit: Retries through a DelegatingHandler would be more elegant solution for the problem because would decouple the code from the retry logic which would make this class a little bit simpler.

Reply:
Yeah, agreed. I think I'll go ahead and take a stab at doing it that way instead for compare/contrast purposes.

Reply:
I've taken a stab at this here: https://github.com/lemillermicrosoft/semantic-kernel/pull/2

The DelegatingHandler cannot be shared, so this also uses a Factory interface to aid with creation of DelegatingHandler instances.